twikit/x_client_transaction/transaction.py

```
@@ -142,10 +160,18 @@ def generate_transaction_id(self, method: str, path: str, response=None, key=Non
         time_now = time_now or math.floor(
             (time.time() * 1000 - 1682924400 * 1000) / 1000)
         time_now_bytes = [(time_now >> (i * 8)) & 0xFF for i in range(4)]
-        key = key or self.key or self.get_key(response)
+        key = key or getattr(self, 'key', None)
+        if not key:
+            try:
+                key = self.get_key(response)
+            except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)